### DISCUSSION TITLE
Development server not spinning up

### DISCUSSION THREAD

**Discussion by b-meet** [vercel/next.js#40828] [Help]
I started learning next.js a few days back form the [official learning course](https://nextjs.org/learn/foundations/about-nextjs). I completed the foundation course. 

I started to create my [first next app](https://nextjs.org/learn/basics/create-nextjs-app), followed the instructions as per the course.

I had the starter file structure in vs-code, then I added `npm run dev` command in integrated terminal. The development server ran and terminal showed this message **_ready - started server on 0.0.0.0:3000, url: http://localhost:3000_** but with a few seconds of running the development server stopped and I was unable to connect to the project on localhost:3000.

I am absolute beginner to nextjs there maybe a silly mistake somewhere but I have followed the official course step by step and still the issue persists. also I tried to recreate the project but facing the same issue again and again.

**Comment by icyJoseph** [comment]
There's a few moving parts here.

- The VSCode integrated terminal might have ran into an issue of its own, did it log anything when the server stopped?
- Did you perhaps press CTRL+c while focused on the terminal?
- Were you able to start it up again?
- Are you using Windows? Have you got all node_modules installed?
- What Node.js and NPM are you using?

This doesn't have much to do with Next.js itself but with NPM and CLI commands. If you could provide more details we can sort this out.

**Reply by b-meet** [reply]
- No the development server just stopped and terminal did not log anything
- Nope I started the dev server as I always do while using `_reactjs_`
- I am able to start the dev server again but facing the same issue
- I am using windows 10. As far as node_modules are considered All the required node_modules should be install when I run `create-next-app` but I also tried npm install again everything was up-to-date. So yes I have all node_modules
- nodejs version 16.16.0
- npm version 8.11.0

If you need any other details please let me know because I am not able to run any nextjs project on my machine.

**Reply by icyJoseph** [reply]
Wait, so you run `npm run dev` the server runs for a few seconds and then shuts down? Would it be possible for you to record your screen and share a video/gif of how it happens? 

I fear this might be some kind of protection from Windows 10, and that it silently shuts down Node.js processes. 

We can try to discard that, if you are able to run anything else at all with `node index.js` in your machine?

Like:

```js
// index.js
function sleep() {
    return new Promise((next) => {
        setTimeout(next, 10_000);
    }).then(() => console.log('woke up'));
}

sleep();
```

Or perhaps you can use `Create-React-App` without a problem? Again I want to make sure this is not a system wide issue with Windows.

**Reply by b-meet** [reply]
`create-react-app` projects are working fine for me this issue is only occurring when I am creating projects using `create-next-app` as far as I know

still to make things clear I will add a short video of what's happening.

[ef88b934-0497-4ac6-ab10-ce3e4399da0c.webm](https://user-images.githubusercontent.com/82386196/191939418-9cbcdfa5-d355-4227-8838-4697e12bdcef.webm)

**Reply by icyJoseph** [reply]
Alright, before narrowing down this further, could you please try these:

- `npx next dev`
- `npx next build`

And if they fail, do `npm install next@latest`, and repeat

Not sure if PowerShell might have anything to do with it though, I'll look it up in the meantime.

**Reply by b-meet** [reply]
`npx next dev` failed to spin up the dev sever. `npx next build` created optimized build

This is what terminal display after running the commands.
![image](https://user-images.githubusercontent.com/82386196/192196567-7924b179-e3e7-456a-9927-90f3f733a478.png)

**Reply by b-meet** [reply]
@icyJoseph FYI: the project is now running. 

I just uninstalled node js form my machine and reinstalled (node v-16.17.1) it, then the development server started and now I am able to see the output on `localhost:3000`. still don't know what was the real issue seems like problem with node.

again there might be some other issue as well.

**Reply by icyJoseph** [reply]
Fascinating stuff. 

I see that `next build` never completed either, it just informed about the work it was about to do and quit. 

Also frustrating that the issue sort of hides under the rough, and the solutions is ¯\\_(ツ)_/¯, but it guess it'll have to do... I'd recommend using WSL(2) for Windows to avoid this sort of complications, or perhaps work with a Docker Image instead.
